src/vscode-dts/vscode.proposed.lmTools.d.ts

```diff
@@ -93,7 +93,7 @@ declare module 'vscode' {
 		 * Invoke a tool with the given parameters.
 		 * TODO@API Could request a set of contentTypes to be returned so they don't all need to be computed?
 		 */
-		export function invokeTool(name: string, parameters: Object, token: CancellationToken): Thenable<LanguageModelToolResult>;
+		export function invokeTool(name: string, parameters: Object, token: CancellationToken): Thenable<LanguageModelToolResult & Disposable>;
```

Comment:
Having to dispose the result is pretty annoying, I guess the root reason for that is that the element could be rendered multiple times, right?

Reply:
It's because the tools side needs to keep a reference to the returned element in case the consumer wants to render it.

Reply:
Yes, that's what this implementation is doing which allows a consumer to return something like `{ myType: <MyPromptElement /> }` as their tool result.

I mentioned my preference for a serializable version in the PR description